projects/storefrontapp-e2e-cypress/cypress/support/vendor/cds/merchandising-carousel.commands.ts

```diff
@@ -0,0 +1,27 @@
+declare global {
+  namespace Cypress {
+    interface Chainable {
+      /**
+       * Wait for a carousel event to be sent to CDS. Returns the sent event
+       *
+       * @memberof Cypress.Chainable
+       *
+       * @example
+        ```
+        cy.waitForCarouselEvent(eventSchema);
+        ```
+       */
+      waitForCarouselEvent: (eventSchema: string) => Cypress.Chainable<{}>;
+    }
+  }
+}
+
+Cypress.Commands.add('waitForCarouselEvent', (eventSchema: string) => {
+  cy.wait(`@carouselEventApiRequest`).then(({ request }) => {
+    if (request.headers['hybris-schema'] !== eventSchema) {
+      return cy.waitForCarouselEvent(eventSchema);
```

Comment:
I think this is flaky 

did you test with dev17 because I kept having consisting failure related to this and others
I will dive deeper and find out the best way

Just a reminder why the need for this function again